### PR TITLE
log-received-messages and log-sent-messages, #21506

### PR DESCRIPTION
**Pull request by patriknw**
- also duplicate the trusted settings to artery section

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #21506

File: akka-remote/src/main/scala/akka/remote/artery/Codecs.scala [hunk 1]
```diff
@@ -89,6 +90,12 @@ private[remote] class Encoder(
 
       override protected def logSource = classOf[Encoder]
 
+      private var debugLogEnabled = false
+
+      override def preStart(): Unit = {
+        debugLogEnabled = log.isDebugEnabled
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/Codecs.scala]: couldn't this do the check of the other boolean as well, as they wont change after this?

File: akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala [hunk 2]
```diff
@@ -27,58 +28,66 @@ private[akka] class MessageDispatcher(
 
   private val remoteDaemon = provider.remoteDaemon
   private val log = Logging(system, getClass.getName)
+  private val debugLogEnabaled = log.isDebugEnabled
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala]: spelling

File: akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala [hunk 3]
```diff
@@ -27,58 +28,66 @@ private[akka] class MessageDispatcher(
...
     recipient match {
 
       case `remoteDaemon` ⇒
-        if (UntrustedMode) log.debug("dropping daemon message in untrusted mode")
-        else {
-          if (LogReceive) log.debug("received daemon message {}", msgLog)
+        if (UntrustedMode) {
+          if (debugLogEnabaled) log.debug(
+            "dropping daemon message [{}] in untrusted mode",
+            messageClassName(message))
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala]: Wouldn't a from host be nice and useful here as well?

**Review by johanandren — APPROVED** [review]
LGTM with some minor things

File: akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala (hunk 3 above)
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/MessageDispatcher.scala]: We don't have that here, but we can probably pass it in from the stage that is calling this. Good idea.

**Comment by drewhk** [comment]
LGTM

**Comment by patriknw** [comment]
@johanandren I have incorporated your feedback, see 0634182 and a60fd52

**Comment by johanandren** [comment]
Nice. LGTM!

**Pull request merged**